File: 941-SortArrayByParity/941-SortArrayByParity.js
// Last updated: 3/7/2026, 10:02:31 PM
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortArrayByParity = function(nums) {
    let i=0; 
    let j=nums.length-1;
    let temp;

    while(j>=i){
        if(nums[i]%2!=0 && nums[j]%2!=0){
            j--;
            continue;
        }
        if(nums[i]%2==0 && nums[j]%2==0){
            i++;
            continue;
        }
        if(nums[i]%2 != 0 && nums[j]%2==0){
            temp = nums[i];
            nums[i] = nums[j];
            nums[j] = temp;
            i++;
            j--;
            continue;
        }
        i++;
    }
    return nums;
};

File: 941-SortArrayByParity/941-SortArrayByParity.cpp
// Last updated: 8/18/2026, 11:51:56 AM
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
        int i = 0;
        int j = nums.size()-1;

        while(j>i){
            if(nums[i]%2==1 && nums[j]%2==0){
                int temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
                i++;
                j--;
            }
            else if(nums[i]%2==0){
                i++;
            }
            else if(nums[j]%2 == 1){
                j--;
            }
        }
        return nums;
    }
};